### PR TITLE
chore(ci): transfer all GPU CI to hyperstack

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: .github/workflows/gpu_full_multi_gpu_tests.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
               - tfhe/src/c_api/**
               - 'tfhe/docs/**.md'
               - Makefile
-              - '.github/workflows/aws_tfhe_multi_gpu**'
+              - '.github/workflows/**_multi_gpu_tests.yml'
```
Comment (pdroalves) [inline .github/workflows/gpu_full_multi_gpu_tests.yml:52]: Why the `**`?

File: .github/workflows/gpu_full_multi_gpu_tests.yml [hunk 2]
```diff
@@ -126,30 +140,39 @@ jobs:
...
 
+      - name: Check device is detected
+        if: ${{ !cancelled() }}
+        run: nvidia-smi
+
       # No need to test core_crypto and classic PBS in integer since it's already tested on single GPU.
       - name: Run multi-bit CUDA integer tests
         run: |
-          make test_integer_multi_bit_gpu_ci
+          BIG_TESTS_INSTANCE=TRUE make test_integer_multi_bit_gpu_ci
```
Comment (pdroalves) [inline .github/workflows/gpu_full_multi_gpu_tests.yml:150]: I grepped into the repository but couldn't deeply understand how `BIG_TESTS_INSTANCE` affects tests. I suppose this is what selects a larger selection of those, is that right?

File: .github/workflows/gpu_full_multi_gpu_tests.yml [hunk 3]
```diff
@@ -126,30 +140,39 @@ jobs:
...
+    if: ${{ always() && needs.cuda-tests-linux.result != 'skipped' }}
+    continue-on-error: true
+    steps:
+      - name: Send message
         uses: rtCamp/action-slack-notify@4e5fb42d249be6a45a298f3c9543b111b02f7907
         env:
-          SLACK_COLOR: ${{ job.status }}
-          SLACK_MESSAGE: "CUDA AWS multi-GPU tests finished with status: ${{ job.status }}. (${{ env.ACTION_RUN_URL }})"
+          SLACK_COLOR: ${{ needs.cuda-tests-linux.result }}
+          SLACK_MESSAGE: "Multi GPU tests finished with status: ${{ needs.cuda-tests-linux.result }}. (${{ env.ACTION_RUN_URL }})"
```
Comment (pdroalves) [inline .github/workflows/gpu_full_multi_gpu_tests.yml:175]: `Multi-GPU`

File: scripts/integer-tests.sh [hunk 4]
```diff
@@ -130,8 +130,8 @@ fi
 # Override test-threads number to avoid Out-of-memory issues on GPU instances
 if [[ "${backend}" == "gpu" ]]; then
     if [[ "${BIG_TESTS_INSTANCE}" == TRUE ]]; then
-        test_threads=5
-        doctest_threads=5
+        test_threads=8
```
Comment (pdroalves) [inline scripts/integer-tests.sh:133]: Have we tested this for all configurations? I remember having an out-of-memory exception when tried to run with more than 6 threads some time ago.

**Review by pdroalves — APPROVED** [review]
On the limit of my knowledge, this PR seems consistent with what we want to do. I just found a missing "-" and added some questions, but couldn't find anything serious to change.

File: .github/workflows/gpu_full_multi_gpu_tests.yml (hunk 2 above)
Comment (agnesLeroy) [reply .github/workflows/gpu_full_multi_gpu_tests.yml:150]: Yes, you can see the logic in scripts/integer-tests.sh, when the gpu feature is activated we will use:
- 8 threads for tests and doc tests when BIG_TESTS_INSTANCE is true (all H100 test workflows & the multi-GPU workflow)
- 3 threads when BIG_TESTS_INSTANCE is false (the rtx A6000 tests)

The CI passed on all devices with this configuration :slightly_smiling_face:

File: scripts/integer-tests.sh (hunk 4 above)
Comment (agnesLeroy) [reply scripts/integer-tests.sh:133]: All tests passed in the CI so it looks good.

File: .github/workflows/gpu_full_multi_gpu_tests.yml (hunk 1 above)
Comment (agnesLeroy) [reply .github/workflows/gpu_full_multi_gpu_tests.yml:52]: Ah I didn't write the exact name of the workflow, ** is a placeholder for "any character".

File: scripts/integer-tests.sh (hunk 4 above)
Comment (agnesLeroy) [reply scripts/integer-tests.sh:133]: (I triggered approval tests by applying the `approved` label to the PR before you reviewed)

**Pull request merged**